conduit/plugins/processors/filterprocessor/fields/searcher_test.go

```diff
@@ -5,89 +5,142 @@ import (
-
+	searcher, err = MakeFieldSearcher(exp, "made-up-expression-type", sampleExpressionStr, false)
+	require.Error(t, err)
 }
 
 // TestCheckTagExistsAndHasCorrectFunction tests that the check tag exists and function relation works
 func TestCheckTagExistsAndHasCorrectFunction(t *testing.T) {
 	// check that something that doesn't exist throws an error
-	err := checkTagAndExpressionExist(expression.EqualToFilter, "SignedTxnWithAD.SignedTxn.Txn.PaymentTxnFields.LoreumIpsum.SDF")
+	err := checkTagAndExpressionExist(expression.EqualTo, "SignedTxnWithAD.SignedTxn.Txn.PaymentTxnFields.LoreumIpsum.SDF")
 	assert.ErrorContains(t, err, "does not exist in transactions")
 
-	err = checkTagAndExpressionExist(expression.EqualToFilter, "LoreumIpsum")
+	err = checkTagAndExpressionExist(expression.EqualTo, "LoreumIpsum")
 	assert.ErrorContains(t, err, "does not exist in transactions")
 
 	// a made up expression type should throw an error
 	err = checkTagAndExpressionExist("made-up-expression-type", "sgnr")
 	assert.ErrorContains(t, err, "is not supported")
 
-	err = checkTagAndExpressionExist(expression.EqualToFilter, "sgnr")
+	err = checkTagAndExpressionExist(expression.EqualTo, "sgnr")
 	assert.NoError(t, err)
 }
+
+func TestInnerTxnSearch(t *testing.T) {
+	var addr sdk.Address
+	addr[0] = 0x1
+	exp, err := expression.MakeExpression(expression.EqualTo, addr.String(), "")
+	require.NoError(t, err)
+
+	txnWithInnerMatch := sdk.SignedTxnWithAD{
+		ApplyData: sdk.ApplyData{
+			EvalDelta: sdk.EvalDelta{
+				InnerTxns: []sdk.SignedTxnWithAD{
+					{
+						SignedTxn: sdk.SignedTxn{
+							Txn: sdk.Transaction{
+								Header: sdk.Header{
+									Sender: addr,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	{
+		// searchInner: false
+		searcher, err := MakeFieldSearcher(exp, expression.EqualTo, "txn.snd", false)
+		require.NoError(t, err)
+
+		// Provide the matching inner transaction.
+		// It matches with searchInner: false.
+		matches, err := searcher.search(&txnWithInnerMatch.EvalDelta.InnerTxns[0])
+
+		require.NoError(t, err)
+		assert.Equal(t, matches, true)
+
+		// Provide the root txn, no match at the root.
+		// It should have no match with searchInner: false.
+		matches, err = searcher.search(&txnWithInnerMatch)
+
+		// No match on inner txn
+		require.NoError(t, err)
+		assert.Equal(t, matches, false)
+	}
+
+	{
+		// searchInner: true
+		searcher, err := MakeFieldSearcher(exp, expression.EqualTo, "txn.snd", true)
+		require.NoError(t, err)
+
+		// Provide the matching inner transaction.
+		// It matches with searchInner: false.
+		matches, err := searcher.search(&txnWithInnerMatch.EvalDelta.InnerTxns[0])
+
+		require.NoError(t, err)
+		assert.Equal(t, matches, true)
+
+		// Provide the root txn, no match at the root.
+		// It should have no match with searchInner: false.
+		matches, err = searcher.search(&txnWithInnerMatch)
+
+		// No match on inner txn
+		require.NoError(t, err)
+		assert.Equal(t, matches, true)
```

Comment:
Comments seem wrong? It should match on inner here right?

Reply:
Thanks, fixed.